src/main/java/com/example/pokebattlez/battle/controller/Battle.java

```diff
@@ -17,56 +20,66 @@
     private final List<PlayerAction> playerActions = new ArrayList<>();
 
     private BattleService battleService;
-
-    public void setBattleService(BattleService battleService) {
-        this.battleService = battleService;
-    }
+    private static final BattleUtils battleUtils = new BattleUtils(); // Ask about this
 
     public void registerPokemonTeam(Long trainerId, List<Pokemon> pokemon) {
         this.pokemon.put(trainerId, pokemon);
     }
 
-    public void switchPokemonTo(Long trainerId, int position) {
+    public void registerSwitchPokemon(Long trainerId, int position) {
         playerActions.add(PlayerAction.builder()
                 .trainerId(trainerId)
                 .switchToPosition(position)
-                .action(PlayerAction::switchPokemon)
+                .action(Battle::switchPokemon)
                 .priority(1000)
                 .build()
         );
     }
 
-    public void useMoveInPosition(Long trainerId, int position) {
+    public void registerUseMove(Long trainerId, int position) {
         playerActions.add(PlayerAction.builder()
                 .trainerId(trainerId)
                 .moveInPosition(position)
-                .action(PlayerAction::useMove)
+                .action(Battle::useMove)
                 .priority(activePokemon.get(trainerId).getStats().get(Stat.SPEED).getValue())
                 .build()
         );
     }
 
     public void cancelAction(Long trainerId) {
-        playerActions.stream().filter(playerAction -> playerAction.getTrainerId().equals(trainerId)).findFirst().ifPresent(playerActions::remove);
+        playerActions.stream()
+                .filter(playerAction -> playerAction.trainerId.equals(trainerId)).findFirst()
+                .ifPresent(playerActions::remove);
     }
 
-    @Data
-    @Builder
-    private static class PlayerAction {
-        private int priority;
+    private static void switchPokemon(Battle battle, PlayerAction playerAction) {
```

Comment:
Since this is Battle's class method, why is Battle object passed to it? Is it a different Battle?

Reply:
I had to make the method static to use from a reference, so passing it the Battle is in order to access the current instance.